src/main/drivers/adc_stm32g4xx.c

```diff
@@ -0,0 +1,552 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "platform.h"
+
+#ifdef USE_ADC
+
+#include "build/debug.h"
+
+#include "drivers/dma_reqmap.h"
+#include "drivers/io.h"
+#include "drivers/io_impl.h"
+#include "drivers/rcc.h"
+#include "drivers/resource.h"
+#include "drivers/dma.h"
+
+#include "drivers/sensor.h"
+
+#include "drivers/adc.h"
+#include "drivers/adc_impl.h"
+
+#include "pg/adc.h"
+
+// XXX Instance and DMA channel defs will be gone in unified target
```

Comment:
H7 and G4 ADC config is an area that needs some more work, as it requires a method to configure which ADC instance to turn on. Oh, and as a matter or fact, F4 and F7 also have defaults in `adc.h`.

Reply:
The fact that targets that were already supported when Unified Targets were implemented still rely on now unusable defines to configure certain aspects should not mean that we should not make these aspects configurable for new targets.
I think it is preferrable to go directly for Unified Target support with G4, instead of first introducing new legacy targets and then have to touch  and test them again when they are converted to Unified Targets.

Reply:
Hmm... There are couples of ways of making this configurable. Easiest would be to just declare all instances as active (`ADCx` instead of `NULL`), while the hardest and the most complex/complete is to introduce a new CLI command `adc` to be used as `adc <pin resource or  adc channel name> <adc instance> <adc channel>. Thoughts?

Reply:
I haven't looked into the specifics of the G4, but I thought on F4 / F7 all ADCs could use the same pins, so for our use case only enabling one should be enough. Or why would we need more than one?
What is the downside of having them all active?
If we make them configurable, then probably best with an array parameter, e.g. `set adc_enabled = 0, 1, 0` for ADC 2.

Reply:
> I haven't looked into the specifics of the G4, but I thought on F4 / F7 all ADCs could use the same pins,

Not really. If that's the case, we don't need `adcTagMap` array.

>  Or why would we need more than one?

Reply:
> Not really. If that's the case, we don't need `adcTagMap` array.

From the look of it this is pretty much a leftover from F3 - there it is needed as different pins run exclusively on one or another ADC. On F4 / F7, all ADC capable pins seem to work on ADC1 and ADC2, so permanently enabling one or the other should do for us, at least as long as we don't use DMA for the ADC (which I don't think is something that we'd get much value from in the first place).

Reply:
Are you suggesting an ADC rework that doesn't use DMA? That's beyond the scope of this PR.

However, as I reviewed how ADC is initialized, I came to notice that an instance is only activated when the instance can serve a designated pin. This means that we can just initialize all `ADCx` with actual instance values. I'll check if this works.